src/common/snippets/src/lowered/pass/iter_handler.cpp

```diff
@@ -115,6 +115,9 @@ bool TransformInnerSplitLoop::run(LinearIR& linear_ir, LinearIR::constExprIt beg
             offset = offset / inner_loop_work_amount * static_cast<int64_t>(m_tail_size);
         }
         inner_loop_end->set_work_amount(m_tail_size);
+        // Since the loop has work amount equal to increment of outer loop, not broadcasted dimension,
+        // we should set `work_amount_const = true` to avoid rewriting in common loop intiialization passes (for example, `InitLoops`)
+        inner_loop_info->set_work_amount_const(true);
```

Comment:
Discussed offline: should be rewritten in bounds of CVS-143394



> BTW, can we even split loops in dynamism? And if we can't, then shouldn't we disable this transformation entirely?

Only `SplitLoops` registered this pass. Since `SplitLoops` is disabled in dynamic case, this pass also is disabled